libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -3398,6 +3398,18 @@ MAV_RESULT GCS_MAVLINK::handle_preflight_reboot(const mavlink_command_int_t &pac
         }
     }
 
+#if AP_BATTERY_ENABLED
+    // Check if shutdown is requested
+    if (is_equal(packet.param1, 2.0f)) {
```

Comment:
I don't think this should be in PREFLIGHT_REBOOT_SHUTDOWN
I think we need MAV_CMD_BATTERY_CONTROL command, which takes a bitmask of batterys to apply this to, and which allows for both shutdown and startup